desktop/ui/src/main/java/org/datacleaner/windows/OpenAnalysisJobAsTemplateDialog.java

```diff
@@ -97,163 +101,177 @@ public OpenAnalysisJobAsTemplateDialog(WindowContext windowContext, DataCleanerC
+            // The column's path contains also the table name in the path
+            tablePath = columnPath.substring(0, columnDelim);
         }
 
-        for (Entry<String, String> variableEntry : metadata.getVariables().entrySet()) {
-            String id = variableEntry.getKey();
-            String value = variableEntry.getValue();
-            JXTextField textField = WidgetFactory.createTextField("Original: " + value);
-            textField.setText(value);
-            _variableTextFields.put(id, textField);
+        return tablePath;
+    }
+
+    private SourceColumnComboBox createSourceColumnComboBoxForColumn(Map<String,
+            List<SourceColumnComboBox>> sourceColumnComboBoxes, String tablePath, String columnPath) {
+        final SourceColumnComboBox comboBox = new SourceColumnComboBox();
+        comboBox.setEnabled(false);
+        comboBox.setName(columnPath);
+        comboBox.addColumnSelectedListener(col -> {
+            if (col != null) { // make sure all comboboxes in a group use the same table
+                final List<SourceColumnComboBox> comboBoxes = sourceColumnComboBoxes.get(tablePath);
+
+                for (SourceColumnComboBox sameTableComboBox : comboBoxes) {
+                    sameTableComboBox.setModel(_datastore, col.getTable());
+                }
+            }
+
+            refreshOpenButtonVisibility();
+        });
+
+        if (!sourceColumnComboBoxes.containsKey(tablePath)) {
+            sourceColumnComboBoxes.put(tablePath, new ArrayList<>());
         }
 
-        _openButton.setEnabled(false);
-        _datastoreCatalog = configuration.getDatastoreCatalog();
+        return comboBox;
+    }
 
-        final String[] datastoreNames = _datastoreCatalog.getDatastoreNames();
-        // the combobox will contain all datastore names and a null for
-        // "not selected"
-        final String[] comboBoxModel = CollectionUtils.array(new String[1], datastoreNames);
-        _datastoreCombobox = new JComboBox<String>(comboBoxModel);
-        _datastoreCombobox.setEditable(false);
-        _datastoreCombobox.addActionListener(new ActionListener() {
-            @Override
-            public void actionPerformed(ActionEvent e) {
-                final String datastoreName = (String) _datastoreCombobox.getSelectedItem();
-                _datastore = _datastoreCatalog.getDatastore(datastoreName);
+    private JComboBox createDatastoreCombobox() {
+        final String[] comboBoxModel = CollectionUtils.array(new String[1], _datastoreCatalog.getDatastoreNames());
+        final JComboBox comboBox = new JComboBox<>(comboBoxModel);
+        comboBox.setEditable(false);
+        final JDialog parent = this;
+        comboBox.addActionListener(e -> {
+            try {
+                ComboBoxUpdater comboBoxUpdater = new ComboBoxUpdater(parent);
+                comboBoxUpdater.execute();
+            } catch (Exception exception) {
+                logger.error(exception.getMessage());
```

Comment:
Are you sure it is good enough if you just log an error here, or should this be propagated to the user interface in some manner?